From: Maxim Dounin <mdounin@mdounin.ru>
Date: Tue, 24 Apr 2018 15:28:58 +0300
Subject: Mail: fixed error message about missing ssl_certificate_key.

In 51e1f047d15d, the "ssl" directive name was incorrectly hardcoded
in the error message shown when there are some SSL keys defined, but
not for all certificates.  Right approach is to use the "mode" variable,
which can be either "ssl" or "starttls".

--- a/src/mail/ngx_mail_ssl_module.c
+++ b/src/mail/ngx_mail_ssl_module.c
@@ -350,10 +350,10 @@ ngx_mail_ssl_merge_conf(ngx_conf_t *cf,
             ngx_log_error(NGX_LOG_EMERG, cf->log, 0,
                           "no \"ssl_certificate_key\" is defined "
                           "for certificate \"%V\" and "
-                          "the \"ssl\" directive in %s:%ui",
+                          "the \"%s\" directive in %s:%ui",
                           ((ngx_str_t *) conf->certificates->elts)
                           + conf->certificates->nelts - 1,
-                          conf->file, conf->line);
+                          mode, conf->file, conf->line);
             return NGX_CONF_ERROR;
         }
 
